---
permalink: /doc/samtools-mpileup.1.html
layout: manpage
title: samtools-mpileup(1) manual page
package: samtools-1.9
date: 14 August 2018
section_text: Bioinformatics tools
---
<h1 id="NAME"><a href="#NAME">NAME</a></h1>
samtools mpileup &ndash; produces "pileup" textual format from an alignment
<h1 id="SYNOPSIS"><a href="#SYNOPSIS">SYNOPSIS</a></h1>
<p>
samtools <a href="samtools-mpileup.1.html">mpileup</a>
[<b>-EB</b>]
[<b>-C</b>
<em>capQcoef</em>]
[<b>-r</b>
<em>reg</em>]
[<b>-f</b>
<em>in.fa</em>]
[<b>-l</b>
<em>list</em>]
[<b>-Q</b>
<em>minBaseQ</em>]
[<b>-q</b>
<em>minMapQ</em>]
<em>in.bam</em>
[<em>in2.bam</em>
[<em>...</em>]]
<p>
<h1 id="DESCRIPTION"><a href="#DESCRIPTION">DESCRIPTION</a></h1>
<p>
Generate pileup for one or multiple BAM files. Alignment records
are grouped by sample (SM) identifiers in @RG header lines. If sample
identifiers are absent, each input file is regarded as one sample.
<p>
Samtools mpileup can still produce VCF and BCF output, but this feature is
deprecated and will be removed in a future release.  Please use
<b>bcftools mpileup</b>
for this instead.  (Documentation on the deprecated options has been removed
from this manual page, but older versions are available online
at &lt;<a href="http://www.htslib.org/doc/">http://www.htslib.org/doc/</a>&gt;.)
<p>
In the pileup format (without
<b>-u</b> or<b> -g</b>),
each
line represents a genomic position, consisting of chromosome name,
1-based coordinate, reference base, the number of reads covering the site,
read bases, base qualities and alignment
mapping qualities. Information on match, mismatch, indel, strand,
mapping quality and start and end of a read are all encoded at the read
base column. At this column, a dot stands for a match to the reference
base on the forward strand, a comma for a match on the reverse strand,
a '&gt;' or '&lt;' for a reference skip, `ACGTN' for a mismatch on the forward
strand and `acgtn' for a mismatch on the reverse strand. A pattern
`\\+[0-9]+[ACGTNacgtn]+' indicates there is an insertion between this
reference position and the next reference position. The length of the
insertion is given by the integer in the pattern, followed by the
inserted sequence. Similarly, a pattern `-[0-9]+[ACGTNacgtn]+'
represents a deletion from the reference. The deleted bases will be
presented as `*' in the following lines. Also at the read base column, a
symbol `^' marks the start of a read. The ASCII of the character
following `^' minus 33 gives the mapping quality. A symbol `$' marks the
end of a read segment.
<p>
Note that there are two orthogonal ways to specify locations in the
input file; via <b>-r</b> <em>region</em> and <b>-l</b> <em>file</em>.  The
former uses (and requires) an index to do random access while the
latter streams through the file contents filtering out the specified
regions, requiring no index.  The two may be used in conjunction.  For
example a BED file containing locations of genes in chromosome 20
could be specified using <b>-r 20 -l chr20.bed</b>, meaning that the
index is used to find chromosome 20 and then it is filtered for the
regions listed in the bed file.
<p>
<h1 id="OPTIONS"><a href="#OPTIONS">OPTIONS</a></h1>
<dl><dt><b>-6, --illumina1.3+</b></dt><dd><p>
Assume the quality is in the Illumina 1.3+ encoding.
</dd><dt><b>-A, --count-orphans</b></dt><dd><p>
Do not skip anomalous read pairs in variant calling.
</dd><dt><b>-b, --bam-list</b><em> FILE</em></dt><dd><p>
List of input BAM files, one file per line [null]
</dd><dt><b>-B, --no-BAQ</b></dt><dd><p>
Disable base alignment quality (BAQ) computation.
See
<b>BAQ</b>
below.
</dd><dt><b>-C, --adjust-MQ</b><em> INT</em></dt><dd><p>
Coefficient for downgrading mapping quality for reads containing
excessive mismatches. Given a read with a phred-scaled probability q of
being generated from the mapped position, the new mapping quality is
about sqrt((INT-q)/INT)*INT. A zero value disables this
functionality; if enabled, the recommended value for BWA is 50. [0]
</dd><dt><b>-d, --max-depth</b><em> INT</em></dt><dd><p>
At a position, read maximally
<em>INT</em>
reads per input file. Setting this limit reduces the amount of memory and
time needed to process regions with very high coverage.  Passing zero for this
option sets it to the highest possible value, effectively removing the depth
limit. [8000]
<p>
Note that up to release 1.8, samtools would enforce a minimum value for
this option.  This no longer happens and the limit is set exactly as
specified.
</dd><dt><b>-E, --redo-BAQ</b></dt><dd><p>
Recalculate BAQ on the fly, ignore existing BQ tags.
See
<b>BAQ</b>
below.
</dd><dt><b>-f, --fasta-ref</b><em> FILE</em></dt><dd><p>
The
<b>faidx</b>-indexed
reference file in the FASTA format. The file can be optionally compressed by
<b>bgzip</b>.
[null]
<p>
Supplying a reference file will enable base alignment quality calculation
for all reads aligned to a reference in the file.  See
<b>BAQ</b>
below.
</dd><dt><b>-G, --exclude-RG</b><em> FILE</em></dt><dd><p>
Exclude reads from readgroups listed in FILE (one @RG-ID per line)
</dd><dt><b>-l, --positions</b><em> FILE</em></dt><dd><p>
BED or position list file containing a list of regions or sites where
pileup or BCF should be generated. Position list files contain two
columns (chromosome and position) and start counting from 1.  BED
files contain at least 3 columns (chromosome, start and end position)
and are 0-based half-open.
<br>
While it is possible to mix both position-list and BED coordinates in
the same file, this is strongly ill advised due to the differing
coordinate systems. [null]
</dd><dt><b>-q, -min-MQ</b><em> INT</em></dt><dd><p>
Minimum mapping quality for an alignment to be used [0]
</dd><dt><b>-Q, --min-BQ</b><em> INT</em></dt><dd><p>
Minimum base quality for a base to be considered [13]
</dd><dt><b>-r, --region</b><em> STR</em></dt><dd><p>
Only generate pileup in region. Requires the BAM files to be indexed.
If used in conjunction with -l then considers the intersection of the
two requests.
<em>STR</em>
[all sites]
</dd><dt><b>-R, --ignore-RG</b></dt><dd><p>
Ignore RG tags. Treat all reads in one BAM as one sample.
</dd><dt><b>--rf, --incl-flags</b><em> STR|INT</em></dt><dd><p>
Required flags: skip reads with mask bits unset [null]
</dd><dt><b>--ff, --excl-flags</b><em> STR|INT</em></dt><dd><p>
Filter flags: skip reads with mask bits set
[UNMAP,SECONDARY,QCFAIL,DUP]
</dd><dt><b>-x, --ignore-overlaps</b></dt><dd><p>
Disable read-pair overlap detection.
</dd><dt><b>-X</b></dt><dd><p>
Include customized index file as a part of arugments. See
<b>EXAMPLES</b>
section for sample of useage.
<p>
</dd></dl><p>
<b>Output Options:</b>
<dl><dt><b>-o, --output </b><em>FILE</em></dt><dd><p>
Write pileup output to
<em>FILE</em>,
rather than the default of standard output.
<p>
(The same short option is used for both the deprecated
<b>--open-prob</b>
option and
<b>--output .</b>
If
<b>-o</b>'s
argument contains any non-digit characters other than a leading + or - sign,
it is interpreted as
<b>--output</b>.
Usually the filename extension will take care of this, but to write to an
entirely numeric filename use
<b>-o ./123</b>
or
<b>--output 123</b>.)
</dd><dt><b>-O, --output-BP</b></dt><dd><p>
Output base positions on reads.
</dd><dt><b>-s, --output-MQ</b></dt><dd><p>
Output mapping quality.
</dd><dt><b>--output-QNAME</b></dt><dd><p>
Output an extra column containing comma-separated read names.
</dd><dt><b>-a</b></dt><dd><p>
Output all positions, including those with zero depth.
</dd><dt><b>-a -a, -aa</b></dt><dd><p>
Output absolutely all positions, including unused reference sequences.
Note that when used in conjunction with a BED file the -a option may
sometimes operate as if -aa was specified if the reference sequence
has coverage outside of the region specified in the BED file.
</dd></dl><p>
<b>BAQ (Base Alignment Quality)</b>
<p>
BAQ is the Phred-scaled probability of a read base being misaligned.
It greatly helps to reduce false SNPs caused by misalignments.
BAQ is calculated using the probabilistic realignment method described
in the paper &ldquo;Improving SNP discovery by base alignment quality&rdquo;,
Heng Li, Bioinformatics, Volume 27, Issue 8
&lt;<a href="https://doi.org/10">https://doi.org/10</a>.1093/bioinformatics/btr076&gt;
<p>
BAQ is turned on when a reference file is supplied using the
<b>-f</b>
option.  To disable it, use the
<b>-B</b>
option.
<p>
It is possible to store pre-calculated BAQ values in a SAM BQ:Z tag.
Samtools mpileup will use the precalculated values if it finds them.
The
<b>-E</b>
option can be used to make it ignore the contents of the BQ:Z tag and
force it to recalculate the BAQ scores by making a new alignment.
<p>
<h1 id="EXAMPLES"><a href="#EXAMPLES">EXAMPLES</a></h1>
<ul><li>Call SNPs and short INDELs:
<pre>
samtools mpileup -uf ref.fa aln.bam | bcftools call -mv &gt; var.raw.vcf
bcftools filter -s LowQual -e '%QUAL&lt;20 || DP&gt;100' var.raw.vcf  &gt; var.flt.vcf
</pre>
The
<b>bcftools filter</b>
command marks low quality sites and sites with the read depth exceeding
a limit, which should be adjusted to about twice the average read depth
(bigger read depths usually indicate problematic regions which are
often enriched for artefacts).  One may consider to add
<b>-C50</b>
to
<b>mpileup</b>
if mapping quality is overestimated for reads containing excessive
mismatches. Applying this option usually helps
<b>BWA-short</b>
but may not other mappers.
<p>
Individuals are identified from the
<b>SM</b>
tags in the
<b>@RG</b>
header lines. Individuals can be pooled in one alignment file; one
individual can also be separated into multiple files. The
<b>-P</b>
option specifies that indel candidates should be collected only from
read groups with the
<b>@RG-PL</b>
tag set to
<em>ILLUMINA</em>.
Collecting indel candidates from reads sequenced by an indel-prone
technology may affect the performance of indel calling.
<p>
</li>
<li>Generate the consensus sequence for one diploid individual:
<pre>
samtools mpileup -uf ref.fa aln.bam | bcftools call -c | vcfutils.pl vcf2fq &gt; cns.fq
</pre>
</li>
<li>Include customized index file as a part of arugments.
<pre>
samtools mpileup [options] -X /data_folder/in1.bam [/data_folder/in2.bam [...]] /index_folder/index1.bai [/index_folder/index2.bai [...]]
</pre>
</li>
<li>Phase one individual:
<pre>
samtools calmd -AEur aln.bam ref.fa | samtools phase -b prefix - &gt; phase.out
</pre>
The
<b>calmd</b>
command is used to reduce false heterozygotes around INDELs.
<p>
</ul>
<h1 id="AUTHOR"><a href="#AUTHOR">AUTHOR</a></h1>
<p>
Written by Heng Li from the Sanger Institute.
<p>
<h1 id="SEE_ALSO"><a href="#SEE_ALSO">SEE ALSO</a></h1>
<a href="samtools.1.html"><em>samtools</em></a> (1),
<a href="samtools-depth.1.html"><em>samtools-depth</em></a> (1),
<a href="samtools-sort.1.html"><em>samtools-sort</em></a> (1),
<a href="bcftools.1.html"><em>bcftools</em></a> (1)
<p>
Samtools website: &lt;<a href="http://www.htslib.org/">http://www.htslib.org/</a>&gt;
